src-electron/generator/helper-zcl.js

```diff
@@ -2864,3 +3124,5 @@ exports.as_type_max_value = as_type_max_value
 exports.as_type_min_value = as_type_min_value
 exports.as_zcl_type_size = as_zcl_type_size
 exports.if_compare = if_compare
+exports.as_underlying_language_specific_zcl_type =
```

Comment:
This feels like it should be handled via overridables like the existing basic_types stuff...  Having it all hardcoded here is odd, since different things may want different type mappings.

Reply:
They seem pretty hard coded in https://github.com/project-chip/zap/blob/master/src-electron/generator/matter/controller/java/templates/helper.js . This should help get all of the dataType and zclType logic to one spot and deprecate the others.

Reply:
If these are Matter-Java-codegen-specific, why are we using a generic name for the helper?

Reply:
To give more context. I am handling things like `{{asJavaBoxedType type chipType}}`, `{{convertAttributeCallbackTypeToJavaName chipCallback.type}}`, `{{asPythonType chipType}}` and `{{chipCallback.name}}` from within stateless helpers such as `zcl_server_attributes` and `enabled_attributes_for_cluster_and_side` instead of `chip_server_cluster_attributes`. That is the reason I created these language specific zcl type helpers and felt having all of these in one place is a good cleanup. I could move these out of this file and put it in the respective files. Is that what you want?

Reply:
>  I could move these out of this file and put it in the respective files.

I would prefer this, yes.

And again, if we can do {{chipCallback.name}} without extra helpers just in the one template that uses it, that would be ideal.....

Reply:
I have moved the code for python and java helpers to the respective files. Need to just figure out on what needs to be done for something like `{{chipCallback.name}}`

Reply:
I think this has too many exceptions to create the right output in the template and can lead to comparing a lot of that in the template. For now I have clubbed this into a helper.